### ISSUE TITLE
Can't call shuffle_batch with a boolean tensor

### ISSUE THREAD

**Issue by nikitakit** [tensorflow/tensorflow#646]
`tf.bool` is not a valid dtype for tensors passed to `tf.train.shuffle_batch` (or most queues, I think)

I'm porting a data input pipeline to tensorflow, but doing it a little differently than the seq2seq tutorials. I have variable-length sequential data that I pad in pre-processing, while also generating a boolean mask for the valid entries. However, I can't store the mask in a `tf.bool` tensor because that breaks the queue system.

**Comment by mrry** [comment]
Thanks for pointing this out! There's no technical reason why this should be, just an oversight on our part. I'll prepare a patch.

**Comment by mrry** [comment]
Should be fixed at HEAD now.
